src/Sulu/Component/Rest/ListBuilder/Metadata/Provider/MetadataProvider.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Metadata\Provider;
+
+use Metadata\MetadataFactoryInterface;
+use Sulu\Component\Rest\ListBuilder\Metadata\ProviderInterface;
+
+/**
+ * Provides metadata with the given metadata-factory.
+ */
+class MetadataProvider implements ProviderInterface
+{
+    /**
+     * @var MetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    public function __construct(MetadataFactoryInterface $metadataFactory)
+    {
+        $this->metadataFactory = $metadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMetadataForClass($className)
+    {
+        return $this->metadataFactory->getMetadataForClass($className);
```

Comment:
This class has only one method, which is delegating to the only injected dependency. Do we really need this class then? What is the intention of this?